### PR TITLE
feat: Optimize highlighted rendering performance in the box

### PR DESCRIPTION
**Pull request by stariveer**
Previously, we used full-scene rendering, where all operations had to iterate through and render the entire scene. Each render took more than 10 seconds in a scene with 100 boxes.

We adopted a highlighted rendering algorithm with a time complexity of O(n * m * k), where n is the number of points, m is the length of the cube list, and k is the number of vertices in the polygon.

After the initial rendering, the color data of the point cloud rendering is cached.

For the most frequent operations (single-box operations), such as dragging a box to adjust its height, size, and position, or adjusting the orientation of the box, each operation only re-renders the following parts:

• The box that has been changed
• The area covered by the box before and after it is moved or resized
• Other boxes that intersect with this area

In this way, m in O(n * m * k) is reduced from 100 (full scene) to less than 5.

For operations that do not involve movement, such as creating a box, deleting a box, switching the main attribute, or toggling the validity of a single box, m = 1.

In summary, for a typical single-frame scene with 100 boxes, the rendering time of high-frequency operations is reduced from 10000～12000ms to around 500～800ms, which is a speed improvement of more than 10 times, greatly enhancing the operation experience.

### CODE REVIEW

File: packages/lb-annotation/src/core/pointCloud/index.ts [hunk 1]
```diff
@@ -1084,30 +1084,59 @@ export class PointCloud extends EventListener {
    * @param boxParams
    * @returns
    */
-  public async highlightOriginPointCloud(pointCloudBoxList?: IPointCloudBox[], highlightIndex: number[] = []) {
+  public async highlightOriginPointCloud(
+    pointCloudBoxList?: IPointCloudBox[],
+    highlightIndex: number[] = [],
+    config?: {
```
Comment (lihqi) [inline packages/lb-annotation/src/core/pointCloud/index.ts:1090]:
```javascript
    config: {
      modifiedBoxIds: string[];
      resetAreas: ICoordinate[][];
    } = {
        modifiedBoxIds: [],
        resetAreas: [],
      },
````
是想给一个默认值么？

**Review by lihqi — CHANGES_REQUESTED** [review]

File: packages/lb-annotation/src/core/pointCloud/index.ts (hunk 1 above)
Comment (stariveer) [reply packages/lb-annotation/src/core/pointCloud/index.ts]: done

File: packages/lb-annotation/src/core/pointCloud/highlightWorker.js [hunk 2]
```diff
@@ -4,6 +4,20 @@
  * @param {*} z
  * @returns
  */
+
```
[the remark this answers is not in the document]
Comment (stariveer) [reply packages/lb-annotation/src/core/pointCloud/highlightWorker.js]: done

**Review by lihqi — APPROVED** [review]
